invenio_accounts/models.py

```diff
@@ -90,6 +121,44 @@ class User(db.Model, Timestamp, UserMixin):
         "LoginInformation", back_populates="user", uselist=False, lazy="joined"
     )
 
+    @hybrid_property
+    def user_profile(self):
+        """Get the user profile."""
+        # NOTE: accessing this property requires an initialized app for config
+        if self._user_profile is None:
+            return None
+        elif not isinstance(self._user_profile, UserProfileDict):
+            self._user_profile = UserProfileDict(**self._user_profile)
+
+        return self._user_profile
+
+    @user_profile.setter
+    def user_profile(self, value):
+        """Set the user profile."""
+        if value is None:
+            self._user_profile = None
+        else:
+            self._user_profile = UserProfileDict(**value)
```

Comment:
when we decide to remove the old user profile table (`user.profile`), we could create a [synonym](https://docs.sqlalchemy.org/en/14/orm/mapped_attributes.html#synonyms) for the new field... and we need to move all the data from the old tables into this new JSON field (via alembic, i suppose)